pkg/query-service/app/http_handler.go

```diff
@@ -1858,3 +1860,120 @@ func (aH *APIHandler) writeJSON(w http.ResponseWriter, r *http.Request, response
 	w.Header().Set("Content-Type", "application/json")
 	w.Write(resp)
 }
+
+// logs
+func (aH *APIHandler) RegisterLogsRoutes(router *mux.Router) {
+	subRouter := router.PathPrefix("/api/v1/logs").Subrouter()
+	subRouter.HandleFunc("", ViewAccess(aH.getLogs)).Methods(http.MethodGet)
+	subRouter.HandleFunc("/tail", ViewAccess(aH.tailLogs)).Methods(http.MethodGet)
+	subRouter.HandleFunc("/fields", ViewAccess(aH.logFields)).Methods(http.MethodGet)
+	subRouter.HandleFunc("/fields", ViewAccess(aH.logFieldUpdate)).Methods(http.MethodPost)
+	subRouter.HandleFunc("/aggregate", ViewAccess(aH.logAggregate)).Methods(http.MethodGet)
+}
+
+func (aH *APIHandler) logFields(w http.ResponseWriter, r *http.Request) {
+	fields, apiErr := (*aH.reader).GetLogFields(r.Context())
+	if apiErr != nil {
+		respondError(w, apiErr, "Failed to fetch fields from the DB")
+		return
+	}
+	aH.writeJSON(w, r, fields)
+}
+
+func (aH *APIHandler) logFieldUpdate(w http.ResponseWriter, r *http.Request) {
+	field := model.UpdateField{}
+	if err := json.NewDecoder(r.Body).Decode(&field); err != nil {
+		apiErr := &model.ApiError{Typ: model.ErrorBadData, Err: err}
+		respondError(w, apiErr, "Failed to decode payload")
+		return
+	}
+
+	err := logs.ValidateUpdateFieldPayload(&field)
+	if err != nil {
+		apiErr := &model.ApiError{Typ: model.ErrorBadData, Err: err}
+		respondError(w, apiErr, "Incorrect payload")
+		return
+	}
+
+	apiErr := (*aH.reader).UpdateLogField(r.Context(), &field)
+	if apiErr != nil {
+		respondError(w, apiErr, "Failed to update filed in the DB")
+		return
+	}
+	aH.writeJSON(w, r, field)
+}
+
+func (aH *APIHandler) getLogs(w http.ResponseWriter, r *http.Request) {
+	params, err := logs.ParseLogFilterParams(r)
+	if err != nil {
+		apiErr := &model.ApiError{Typ: model.ErrorBadData, Err: err}
+		respondError(w, apiErr, "Incorrect params")
+		return
+	}
+
+	res, apiErr := (*aH.reader).GetLogs(r.Context(), params)
+	if apiErr != nil {
+		respondError(w, apiErr, "Failed to fetch logs from the DB")
+		return
+	}
+	aH.writeJSON(w, r, map[string]interface{}{"results": res})
+}
+
+func (aH *APIHandler) tailLogs(w http.ResponseWriter, r *http.Request) {
+	params, err := logs.ParseLogFilterParams(r)
+	if err != nil {
+		apiErr := &model.ApiError{Typ: model.ErrorBadData, Err: err}
+		respondError(w, apiErr, "Incorrect params")
+		return
+	}
+
+	// create the client
+	client := &model.LogsTailClient{Name: r.RemoteAddr, Logs: make(chan *model.GetLogsResponse, 1000), Done: make(chan *bool), Error: make(chan error), Filter: *params}
+	go (*aH.reader).TailLogs(r.Context(), client)
+
+	w.Header().Set("Connection", "keep-alive")
+	w.Header().Set("Content-Type", "text/event-stream")
+	w.Header().Set("Cache-Control", "no-cache")
+	w.Header().Set("Access-Control-Allow-Origin", "*")
```

Comment:
no expertise with http headers. is there some context behind allowing everything here?

Reply:
So we already allow all origins, https://github.com/SigNoz/signoz/blob/f8f903848e914d529617c6e10c69b3644f8d4c30/pkg/query-service/app/server.go#L167 . But I have explicitly added it here again since I am already setting the required headers which are required for event-stream.

Reply:
Sure, have created an issue for it https://github.com/SigNoz/signoz/issues/1494